### PR TITLE
Simon Data Connector

### LINKED ISSUE #3699: Connector for Simon Data
### Is your feature request related to a specific problem?

Simon Data has privacy endpoints that we can use to generate a connector.

[Privacy API Docs](https://documenter.getpostman.com/view/11533956/T17Q6jiE#fef04391-d174-4694-81f2-1e663377ea74)

### Describe the solution you'd like

Use the saas connector template to generate a new connector for Simon Data

### Describe alternatives you've considered, if any

n/a

### Additional context

I think we may still require a dev account here to test with, but will confirm with @Kelsey-Ethyca

### PR DESCRIPTION
**Pull request by SteveDMurphy**
Closes #3699

### Description Of Changes

Used the saas connector template via `nox` :fire:

### Code Changes

* [x] Added config for new connector
* [x] Added dataset for new connector
* [ ] Validated tests for new connector
* [x] Added an icon for the new connector

### Steps to Confirm

* [ ] The connector can be configured and used when processing a DSR

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Comment by SteveDMurphy** [comment]
@galvana - I think I need some help with secrets here potentially? There also appear to be some existing mypy and pytest errors but I think those are known. Thanks for any help here!!

File: CHANGELOG.md
Comment (galvana) [inline CHANGELOG.md:1]: Might need to reconcile the change log, looks like other changes made it in.

File: data/saas/config/simon_data_config.yml
Comment (galvana) [inline data/saas/config/simon_data_config.yml:1]: Perfect ✨

File: tests/fixtures/saas/simon_data_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+@pytest.fixture
+def simon_data_external_references() -> Dict[str, Any]:
+    return {}
+
+
+@pytest.fixture
+def simon_data_erasure_external_references() -> Dict[str, Any]:
+    return {}
```
Comment (galvana) [inline tests/fixtures/saas/simon_data_fixtures.py:43]: You can delete this fixture, `simon_data_external_references`, and remove them as parameters in the `ConnectorRunner` below. Kinda clunky but in terms of the SaaS template nox command it's better to have them and not need them than to need them and not have them.

File: tests/ops/integration_tests/saas/test_simon_data_task.py
Comment (galvana) [inline tests/ops/integration_tests/saas/test_simon_data_task.py:1]: Check out `tests/ops/integration_tests/saas/test_surveymonkey_task.py` for a reference and let's add an assertion to make sure that the first endpoint in the traversal only returns objects that match our identity email. Same thing for the erasures, let's define our expectation of which collections should be affected by the erasure.

File: tests/fixtures/saas/simon_data_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+@pytest.fixture
+def simon_data_erasure_data(
+    simon_data_erasure_identity_email: str,
+) -> Generator:
+    # create the data needed for erasure tests here
+    yield {}
```
Comment (galvana) [inline tests/fixtures/saas/simon_data_fixtures.py:51]: Does Simon Data allow us to submit an erasure request if we don't have any data for the erasure identity email?

File: data/saas/config/simon_data_config.yml
Comment (SteveDMurphy) [reply data/saas/config/simon_data_config.yml:1]: 🤗

**Comment by SteveDMurphy** [comment]
closing in favor of #4552

**Pull request closed without merging**